orchestrator/test_runner/src/unhalt_bridge.rs

```diff
@@ -170,45 +170,16 @@ pub async fn unhalt_bridge_test(
 
     info!("Preparing governance proposal!!");
     // Unhalt the bridge
-    let _ = submit_and_pass_unhalt_bridge_proposal(initial_valid_nonce, contact, &keys)
-        .await
-        .expect("Governance proposal failed");
-    let start = Instant::now();
-    loop {
-        let new_nonces = get_nonces(&mut grpc_client, &keys, &prefix).await;
-        if new_nonces.iter().min() == after_halt_nonces.iter().max() {
-            info!(
-                "Nonces have not changed: {:?}=>{:?} sleeping before retry",
-                after_halt_nonces, new_nonces
-            );
-            if Instant::now()
-                .checked_duration_since(start)
-                .unwrap()
-                .gt(&Duration::from_secs(10 * 60))
-            {
-                panic!("10 minutes have elapsed trying to get the validator last nonces to change for val1 and val2!");
-            }
-            sleep(Duration::from_secs(10)).await;
-            continue;
-        } else {
-            info!("Nonces changed! {:?}=>{:?}", after_halt_nonces, new_nonces);
-            break;
-        }
-    }
-    let mut not_equal = false;
-    while Instant::now() - start < TOTAL_TIMEOUT {
-        let after_unhalt_nonces = get_nonces(&mut grpc_client, &keys, &prefix).await;
-        not_equal = after_unhalt_nonces
-            .iter()
-            .all(|&nonce| nonce == initial_valid_nonce);
-        if not_equal {
-            break;
-        }
-        sleep(Duration::from_secs(1)).await;
-    }
+    submit_and_pass_unhalt_bridge_proposal(initial_valid_nonce, contact, &keys).await;
+    wait_for_proposals_to_execute(contact).await;
```

Comment:
I love the convenience functions available in deep_space, this is awesome that you were able to simplify this giant chunk.